server/src/main/java/org/elasticsearch/action/update/TransportUpdateAction.java

```diff
@@ -333,4 +343,58 @@ private void handleUpdateFailureWithRetry(
         }
         listener.onFailure(cause instanceof Exception ? (Exception) cause : new NotSerializableExceptionWrapper(cause));
     }
+
```

Comment:
Can you add javadoc to explain why this function is needed and what it does. The internal comment is useful but it might be difficult to understand why this is needed.

Reply:
Added, LMKWYT!